BlueprintUI/Sources/Blueprint View/ElementPreview.swift

```diff
@@ -0,0 +1,238 @@
+///
+@available(iOS 13.0, *)
+public struct ElementPreview : View {
+    
+    // MARK: Properties
+    
+    /// A provider which returns a new element.
+    public typealias ElementProvider = () -> Element
+    
+    /// The types of previews to include in the Xcode preview.
+    private var previewTypes : [PreviewType]
+    
+    /// The provider which vends a new element.
+    private var provider : ElementProvider
+    
+    // MARK: Initialization
+    
+    /// Creates a new `ElementPreview` with several common devices that your users may use.
+    public static func commonDevices(with provider : @escaping ElementProvider) -> Self {
+        return Self(
+            with: [
+                .device(.iPhoneSE),
+                .device(.iPhone8),
+                .device(.iPhone8Plus),
+                .device(.iPhoneXs),
+                .device(.iPhoneXsMax),
+                
+                .device(.iPadPro_9_7),
+                .device(.iPadPro_10_5),
+                .device(.iPadPro_12_9_3),
+            ],
+            with: provider
+        )
+    }
+
+    /// Creates a new `ElementPreview` with the provided preview type.
+    /// If you do not pass a preview type, `.thatFits` is used.
+    public init(
+        with previewType : PreviewType = .thatFits,
+        with provider : @escaping ElementProvider
+    ) {
+        self.init(
+            with: [previewType],
+            with: provider
+        )
+    }
+    
+    /// Creates a new `ElementPreview` with the provided preview types.
+    ///
+    /// You can pass as many preview types as you would like to see your
+    /// element rendered in those different environments.
+    ///
+    /// If you do not pass a preview type, `.thatFits` is used.
+    public init(
+        with previewTypes : [PreviewType],
+        with provider : @escaping ElementProvider
+    ) {
+        self.previewTypes = previewTypes
+        self.provider = provider
+    }
+    
+    // MARK: View
+    
+    public var body: some View {
+        ForEach(self.previewTypes, id: \.self) { previewType in
+            previewType.preview(for: ElementView(element: self.provider()))
+        }
+    }
+    
+    private struct ElementView : UIViewRepresentable {
+        
+        var element : Element
+        
+        func makeUIView(context: Context) -> BlueprintView {
+            return BlueprintView(element: self.element)
+        }
+        
+        func updateUIView(_ view: BlueprintView, context: Context) {
+            view.element = self.element
+        }
+    }
+}
+
+
+@available(iOS 13.0, *)
+extension ElementPreview {
+   
+    /// The preview type to use to display an element in an Xcode preview.
+    ///
+    /// We provide three preview types: A specific device type, a fixed size, and the size that fits the view.
+    public enum PreviewType : Hashable {
+        
+        /// The preview will be inside the provided device (eg, iPhone X).
+        case device(Device)
```

Comment:
It’d be nice if this held a `PreviewDevice` and we made the current `Device` cases be `static var` extensions on `PreviewDevice` instead. I think it would be the same API from the call site, but would allow consumers to use new `PreviewDevice`s without having to bump Blueprint.

Reply:
Oh interesting, I like that, will do

Reply:
Done!